cranelift/faerie/src/backend.rs

```diff
@@ -211,8 +211,11 @@ impl Backend for FaerieBackend {
             ref data_decls,
             ref function_relocs,
             ref data_relocs,
+            section: ref datasection,
         } = data_ctx.description();
 
+        assert!(datasection.is_none(), "Custom sections not supported");
```

Comment:
In Rust, we don't use assertions to indicate that the user provided bad input to a library function, instead we use the error type of the function's `Result`. In this case, we want to do something like:
```rust
if let Some((segment, section)) = datasection {
    return Err(cranelift_module::ModuleError::Backend(anyhow::anyhow!("Custom section not supported by cranelift-faerie: `{}:{}`", segment, section)));
}